crates/agglayer-grpc-api/src/node_state_service.rs

```diff
@@ -135,4 +139,201 @@ where
+                    tonic::Code::NotFound,
+                    "Failed to get latest pending certificate",
+                    ErrorDetails::with_error_info(
+                        GetNetworkStatusErrorKind::MissingLatestPendingCertificate.as_str_name(),
+                        GET_NETWORK_STATUS_METHOD_PATH,
+                        [],
+                    ),
+                )
+            })?;
+
+        // Determine network type from the latest available certificate
+        let network_type = match self
+            .service
+            .get_latest_available_certificate_for_network(network_id)
+        {
+            Ok(Some(certificate)) => {
+                // Determine network type based on aggchain_data variant
+                match certificate.aggchain_data {
+                    agglayer_types::aggchain_proof::AggchainData::ECDSA { .. } => "ECDSA",
+                    agglayer_types::aggchain_proof::AggchainData::Generic { .. } => "Generic",
+                }
+            }
+            Ok(None) => {
+                // No certificate found, use default/unknown type
+                warn!(
+                    "No certificate found for network {}, using default network type",
+                    network_id
+                );
+                "Unknown"
+            }
+            Err(error) => {
+                // Error retrieving certificate, log and use default
+                warn!(
+                    ?error,
+                    "Failed to get certificate for network {}, using default network type",
+                    network_id
+                );
+                "Unknown"
+            }
+        };
+
+        // TODO: Define network status. Could represent the healthiness of the network
+        // in regard to the agglayer-node. We could have multiple kind of status
+        // that could represent a network sending too many unprovable certs,
+        // or even a network that didn't settle for N epochs and such (optional).
+        let network_status = "TBD";
+
+        // Extract settled certificate data
+        let (settled_height, settled_cert_id, _settled_epoch) = latest_settled_certificate
+            .as_ref()
+            .map(|cert| {
+                (
+                    cert.height.as_u64(),
+                    Some(cert.certificate_id.into()),
+                    cert.epoch_number,
+                )
+            })
+            .unwrap_or((0, None, None));
+
+        // Get pending certificate error if exists
+        let pending_error = latest_pending_certificate
+            .as_ref()
+            .and_then(|cert| match &cert.status {
+                agglayer_types::CertificateStatus::InError { error } => Some(error.to_string()),
+                _ => None,
+            })
+            .unwrap_or_default();
+
+        // Get epoch with latest settlement from settled certificate header
+        let latest_epoch_with_settlement = latest_settled_certificate
```

Comment:
you already get it ~20 lines above (`_settled_epoch`)

```suggestion
        let latest_epoch_with_settlement = _settled_epoch
```

same thing in the impl on the json rpc side, btw if we can also find a quick way to avoid the duplicated implementation between here and in `json-rpc` api

Reply:
Yes, we will move it to agglayer-rpc.